columnq/src/table/mod.rs

```diff
@@ -1,528 +1,549 @@
-        _ => unreachable!(),
-    };
-
-    let t = if uri == "stdin" {
-        let mut buffer = Vec::new();
-        std::io::stdin().read_to_end(&mut buffer).map_err(|e| {
-            ColumnQError::Generic(format!("Failed to read table data from stdin: {:?}", e))
-        })?;
-        TableSource::new(table_name, TableIoSource::Memory(buffer))
-    } else {
-        TableSource::new(table_name, uri.to_string())
-    };
-
-    // parse extra options from table uri
-    let mut option_json = serde_json::map::Map::new();
-    for opt_str in uri_args.into_iter() {
-        let mut parts = opt_str.splitn(2, '=');
-        let opt_key = parts
-            .next()
-            .ok_or_else(|| ColumnQError::Generic(format!("invalid table option: {:?}", opt_str)))?;
-        let opt_value = parts
-            .next()
-            .ok_or_else(|| ColumnQError::Generic(format!("invalid table option: {:?}", opt_str)))?;
-        option_json.insert(
-            opt_key.to_string(),
-            serde_json::from_str(opt_value).unwrap_or_else(|_| opt_value.into()),
-        );
-    }
-
-    if !option_json.is_empty() {
-        let opt: TableLoadOption = serde_json::from_value(serde_json::Value::Object(option_json))
-            .map_err(|e| {
-            ColumnQError::Generic(format!("Failed to parse table option: {:?}", e))
-        })?;
-        Ok(t.with_option(opt))
-    } else {
-        Ok(t)
-    }
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use pretty_assertions::assert_eq;
-
-    #[test]
-    fn uri_deserialization() -> anyhow::Result<()> {
-        let table_source: TableSource = serde_yaml::from_str(
-            r#"
-name: "ubuntu_ami"
-uri: "test_data/ubuntu-ami.json"
-option:
-  format: "json"
-  pointer: "/aaData"
-  array_encoded: true
-schema:
-  columns:
-    - name: "zone"
-      data_type: "Utf8"
-    - name: "name"
-      data_type: "Utf8"
-"#,
-        )?;
-
-        assert_eq!(
-            table_source.io_source,
-            TableIoSource::Uri("test_data/ubuntu-ami.json".to_string())
-        );
-
-        Ok(())
-    }
-
-    #[test]
-    fn test_parse_table_uri() {
-        let t = parse_table_uri_arg("t=a/b/c").unwrap();
-        assert_eq!(TableSource::new("t", "a/b/c"), t);
-
-        let t = parse_table_uri_arg("t=s3://a/b/c,format=csv,has_header=true").unwrap();
-        assert_eq!(
-            TableSource::new("t", "s3://a/b/c").with_option(TableLoadOption::csv(
-                TableOptionCsv::default().with_has_header(true),
-            )),
-            t
-        );
-
-        let t = parse_table_uri_arg("s3://a/b/foo.csv").unwrap();
-        assert_eq!(TableSource::new("foo", "s3://a/b/foo.csv"), t);
-
-        let t = parse_table_uri_arg("s3://a/b/foo.csv,format=csv").unwrap();
-        assert_eq!(
-            TableSource::new("foo", "s3://a/b/foo.csv")
-                .with_option(TableLoadOption::csv(TableOptionCsv::default())),
-            t
-        );
-    }
-}
+use std::convert::TryFrom;

```

Comment:
@tobyhede the diff in this file looks weird, do you know why every line is considered different by git?

Reply:
Oh i see, there is your editor is using a different line ending encoding (CRLF). Could you change it to LF instead? If you are using visual studio, here is how you can make this change: https://docs.microsoft.com/en-us/visualstudio/ide/encodings-and-line-breaks?view=vs-2022

Reply:
Sorry, I thought I left a comment saying this was still WIP - the diff was only wrong once it was pushed so I didn't pick it up locally. Fixed the line break issue.